src/ILCompiler.Compiler/src/Compiler/NameMangler.cs

```diff
@@ -345,5 +348,40 @@ private string ComputeMangledFieldName(FieldDesc field)
 
             return mangledName;
         }
+
+        private ImmutableDictionary<string, string> _mangledStringLiterals = ImmutableDictionary<string, string>.Empty;
+
+        public string GetMangledStringName(string literal)
+        {
+            string mangledName;
+            if (_mangledStringLiterals.TryGetValue(literal, out mangledName))
+                return mangledName;
+
+            return ComputeMangledStringLiteralName(literal);
+        }
+
+        private string ComputeMangledStringLiteralName(string literal)
+        {
+            string mangledName = SanitizeName(literal);
+
+            if (mangledName.Length > 30)
+                mangledName = mangledName.Substring(0, 30);
+
+            if (mangledName != literal)
+            {
+                if (_sha1 == null)
```

Comment:
(Also, this might be making the compiler non-selfhostable again because on UWP this is forwarded to WinRT APIs and we don't have Win32 replacements yet.)